dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/specific/uploadconfig/elements/SectionUploadConfigBuilder.kt

```diff
@@ -0,0 +1,80 @@
+package org.dataland.frameworktoolbox.specific.uploadconfig.elements
+
+import org.dataland.frameworktoolbox.intermediate.components.support.SelectionOption
+import org.dataland.frameworktoolbox.specific.viewconfig.functional.FrameworkBooleanLambda
+
+/**
+ * An In-Memory representation of a MLDTSectionConfig
+ * @param label the displayed label of the section
+ * @param expandOnPageLoad if true, the section is expanded when it is first rendered
+ * @param shouldDisplay a lambda deciding if this section should be displayed or not
+ * @param children the elements contained in this section
+ * @param labelBadgeColor the color of the badge in which the label is contained
+ */
+
+@Suppress("LongParameterList")
+data class SectionUploadConfigBuilder(
+    override val parentSection: SectionUploadConfigBuilder?,
+    val name: String,
+    var label: String,
+    var expandOnPageLoad: Boolean,
+    var shouldDisplay: FrameworkBooleanLambda,
```

Comment:
Currently iam unsure if FrameworkBooleanLambda is the best name for this type. What do you guys think?

Reply:
Well it contains a lambda body, that returns a boolean, and is used to control the visiblity of cells (`shouldDisplay`) in framework-view-configs.
What other name would you use? Suggestions?

Reply:
True, i do not have a better suggetion right now. I was just wondering, because reading it for the first time i did not understand what a framework boolean is. I would have expected something like display boolean.

Reply:
Closed